packages/ui/src/ui/pages/query-tracker/module/cell-preview/actions.ts

```diff
@@ -27,7 +27,9 @@ export const showQueryTrackerCellPreviewModal = (
                 ),
             );
 
-            const data = (response.rows[0][options.columnName][0] as any[])[0].val;
+            const dataRow = response.rows[0][options.columnName][0] as any | any[];
+
+            const data = Array.isArray(dataRow) ? dataRow[0].val : dataRow.val;
```

Comment:
My concern is `dataRow[0].val`. Won't there be a situation where the array will come empty?